### PR TITLE
Update starknet.js/starknet-devnet/starknet-hardhat-plugin

### PR DESCRIPTION
**Pull request by archseer**
BCI-986

### CODE REVIEW

File: contracts/requirements.txt [hunk 1]
```diff
@@ -2,5 +2,5 @@ ecdsa
 fastecdsa
 sympy
 cairo-lang==0.10.3
-starknet-devnet>=0.4.2
+starknet-devnet>=0.4.6
```
Comment (augustbleeds) [inline contracts/requirements.txt]: Starknet just released https://github.com/Shard-Labs/starknet-devnet/releases 0.5.0a0 which supports starknet 0.11 and cairo 1.0 but I believe right now they only support x86

File: packages-ts/starknet-gauntlet-ocr2/src/lib/encoding.ts [hunk 2]
```diff
@@ -20,26 +20,28 @@ export function bytesToFelts(data: Uint8Array): string[] {
...
+  //   let chunk = felt.toArray('be', Math.min(CHUNK_SIZE, length))
+  //   data.push(...chunk)
 
-    length -= chunk.length
-  }
+  //   length -= chunk.length
+  // }
 
   return Buffer.from(data)
 }
```
Comment (archseer) [inline packages-ts/starknet-gauntlet-ocr2/src/lib/encoding.ts]: Note: Still need to fix this

File: contracts/src/chainlink/solidity/emergency/StarknetValidator.sol [hunk 3]
```diff
@@ -123,15 +133,15 @@ contract StarknetValidator is TypeAndVersionInterface, AggregatorValidatorInterf
...
+  ) external payable override checkAccess returns (bool) {
     return _sendUpdateMessageToL2(currentAnswer);
   }
 
   /**
    * @notice retries to send the latest answer as update message to L2
    * @dev only with access, useful in cases where a previous x-domain message was handeled unsuccessfully.
    */
-  function retry() external checkAccess returns (bool) {
+  function retry() external payable checkAccess returns (bool) {
```
Comment (augustbleeds) [inline contracts/src/chainlink/solidity/emergency/StarknetValidator.sol:144]: Based on the other L2 validators (and the deposit/withdraw function defined here) I don't think the Aggregator is supposed to fund the fees. The validator is meant to withdraw the fees from it's own balance. So I think to get the tests to pass, just deposit some eth to this contract (it has a payable receive() method)

Comment (augustbleeds) [reply contracts/src/chainlink/solidity/emergency/StarknetValidator.sol:144]: Example on Arbitrum https://github.com/smartcontractkit/chainlink/blob/c8d7632bd817ca993e40862d53be7910ff38d34c/contracts/src/v0.8/dev/ArbitrumValidator.sol

Comment (archseer) [reply contracts/src/chainlink/solidity/emergency/StarknetValidator.sol:144]: Hmm could you take care of this in a follow-up PR? I just want to unblock merging the dependency bumps

Comment (archseer) [reply contracts/src/chainlink/solidity/emergency/StarknetValidator.sol:144]: I can also exclude these changes and let it fail on that case for now

**Pull request merged**